spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py

```diff
@@ -36,15 +36,13 @@ def __init__(self, file_path, sampling_frequency, numchan, dtype, recording_chan
         self._geom = geom
 
         if recording_channels is not None:
-            assert len(recording_channels) == self._timeseries.shape[0], \
```

Comment:
this assertion should be satisfied in my opinion. Is there a reason why you removed it?

Reply:
I removed the assertion when I was trying to make the binary data recording extractor support broken channels. In case of broken channels, you want to subsample the number of channels. Because of this, the length of the provided list of channels is not necessarily equal to the true number of channels that are present in the recording. I realize now that it might be better to use a SubRecordingExtractor here.

Reply:
Also in the probe file you can skip channels! It will extract a SubRecordingExtractor automatically

Reply:
@jwouters91 can you add the assertion back?